dynamiq/nodes/agents/react.py

```diff
@@ -1309,13 +1309,39 @@ def _run_agent(
                     self._prompt.messages.append(Message(role=MessageRole.USER, content=observation, static=True))
 
                     if self.streaming.enabled and self.streaming.mode == StreamingMode.ALL:
+                        if "tool" in locals():
+                            source_name = tool.name
+                            tool_name = tool.name
+                        elif isinstance(action, list):
+                            tool_names = []
+                            for tool_data in action:
+                                if isinstance(tool_data, dict) and "name" in tool_data:
+                                    tool_names.append(tool_data["name"])
+                                else:
+                                    tool_names.append("unknown_tool")
```

Comment:
- Please move this name to contants
- This could be rewritten as comprehensive list